stratum/hal/lib/barefoot/bfrt_packetio_manager.cc

```diff
@@ -115,6 +92,8 @@ ::util::Status BfrtPacketioManager::Shutdown() {
     packetout_header_.clear();
     packetin_header_size_ = 0;
     packetout_header_size_ = 0;
+    packet_receive_channel_->Close();
+    pthread_join(sde_rx_thread_id_, nullptr);
```

Comment:
There might be a potential deadlock here.
If you call this Shutdown right after the pipeline config gets pushed, the method will first hold `data_lock_` and the `HandleSdePacketRx` thread will wait for the lock at line 262. And `pthread_join` will never return and stuck.

Reply:
I think one way to solve this issue is to remove `pthread_join` from this block and use another with ReaderMutexLock since we are not modifying the thread id variable
for example:

```cpp
{
  absl::WriterMutexLock l(&data_lock_);
  packetin_header_.clear();
  packetout_header_.clear();
  packetin_header_size_ = 0;
  packetout_header_size_ = 0;
  packet_receive_channel_->Close();
  initialized_ = false;
}
{
  absl::ReaderMutexLock l(&data_lock_);
  pthread_join(sde_rx_thread_id_, nullptr);
}
```